library/src/main/java/com/datatorrent/lib/parser/Parser.java

```diff
@@ -61,25 +70,48 @@ public void setup(PortContext context)
     }
   };
 
-  @OutputPortFieldAnnotation(optional = true)
   public transient DefaultOutputPort<INPUT> err = new DefaultOutputPort<INPUT>();
```

Comment:
yes, thanks Chandni. I will make the type of error port as generics. But I will leave the behavior of emitting of error tuples by default as is for now. We can revisit it later if needed.